src/ast/ast_validations.c

```diff
@@ -593,19 +594,54 @@ static AST_Validation _Validate_CALL_Clauses(const AST *ast) {
 					goto cleanup;
 				}
 			}
+		}
+
+		// if no projections are specified, add the default projections
+		if(proj_count == 0) {
+			uint default_count = Procedure_OutputCount(proc);
+			for(uint j = 0; j < default_count; j ++) {
+				const char *output = Procedure_GetOutput(proc, j);
+				raxInsert(identifiers, (unsigned char *)output,
+						  strlen(output), NULL, NULL);
+			}
+		}
+
+		// collect all entities that are bound before this CALL clause
+		defined_aliases = raxNew();
+		uint clause_idx = clause_indices[i];
+		for(uint j = start_offset; j < clause_idx; j ++) {
+			const cypher_astnode_t *prev_clause = cypher_ast_query_get_clause(
+													  ast->root, j);
+			_AST_GetDefinedIdentifiers(prev_clause, defined_aliases);
+		}
+		start_offset = clause_idx;
```

Comment:
Isn't `start_offset` being cleared at the next iteration `572: uint start_offset = 0;`

Seems like we recompute `defined_aliases` from scratch for each `CALL` clause, this is a bit wasteful, I hope we can enrich it with each iteration, introducing the current `CALL` identifiers in addition to all of the clauses between the current `CALL` clause and the previous one.

Consider: `CALL algo.BFS(1, 2, 3) YIELD nodes CALL algo.BFS(1, 2, 3) YIELD nodes AS n CALL algo.BFS(1, 2, 3) YIELD nodes`

Reply:
You're right; I've updated this logic to match that of `UNWIND` - `defined_aliases` is never cleared and each clause is visited only once.